service/route_handlers.go

```diff
@@ -554,25 +635,83 @@ func HandleSamlAssertion(w http.ResponseWriter, r *http.Request, assertion *saml
 		Path:   "/",
 	}
 	http.SetCookie(w, tokenCookie)
-
-	log.Debugf("redirecting the user with token to %v", redirectURL)
+	log.Debugf("redirecting the user with cookie %v to %v", tokenCookie, redirectURL)
 
 	http.Redirect(w, r, redirectURL, http.StatusFound)
 	return
 }
 
+func PostSamlTokenHTML(w http.ResponseWriter, r *http.Request) {
+	var token, finalRedirectURL string
+	var secure bool
+	var err error
+
+	r.ParseForm()
+
+	if len(r.Form["token"]) > 0 { //array of string
+		token = r.Form["token"][0]
+	}
+
+	if len(r.Form["finalRedirectURL"]) > 0 {
+		finalRedirectURL = r.Form["finalRedirectURL"][0]
+	}
+
+	if len(r.Form["secure"]) > 0 {
+		secure, err = strconv.ParseBool(r.Form["secure"][0])
+		if err != nil {
+			log.Errorf("PostSamlTokenHTML: Error in parsing secure flag for token: %v", err)
+			return
```

Comment:
what would happen to a request if you hit this code block?
Seems like you would just return a blank 200 response? if so, that doesnt seem right. seems like it should 500